Feat/Chore: Add calendar reference and update dependencies

tabedzki opened the pull request:
Introduce a calendar reference to manage event views and update Python and Streamlit dependencies for improved compatibility.

Hi im-peraltiva, 

I modified the calendar to return a view as part of the dict for `eventsSet`. This would allow me to determine when someone has used one of the buttons to change the view. I plan on using this to update the title of events depending on the calendar view. 

Also, in order to develop it, I had to update some of the poetry commands. I also saw messages about outdated packages when using `npm install` and `npm audit fix`. 

I can open a separate PR for the `npm` issues, though it might be better served to retest those after closing some fo the chore(deps) PRs.

Happy to take on some of that work if you'd like

im-perativa commented:
HI @tabedzki thanks for submitting this, could you provide a public streamlit app link to directly test this changes? Also I don't think updating python version is necessary since this changes mostly related to frontend

tabedzki commented:
Happy to provide a working demo online. 

I updated the poetry and Python version since I was unable to use 3.8 because 3.8 is End of Life and 3.9 goes EoL later this year. Therefore, I've updated it for future contributors to not need to adjust the files to get it to run.

tabedzki commented:
@im-perativa
Can you assist me in preparing the component for deployment? I have not worked with TypeScript or node before so this is  new to me.

Steps taken: 
I have run `npm run build` and set the release flag to True, but I am unable to get the package to load properly despite being to when using `Release = False`. 


How does one set up the node for production purposes?

im-perativa commented:
:tada: This issue has been resolved in version 1.3.0 :tada:

The release is available on PyPI and GitHub.
